FuelSDK-CSharp/ETClient.cs

```diff
@@ -110,9 +110,40 @@ public ETClient(NameValueCollection parameters = null, RefreshState refreshState
                 {
                     configSection.Scope = parameters["scope"];
                 }
+                if (parameters.AllKeys.Contains("applicationType"))
+                {
+                    configSection.ApplicationType = parameters["applicationType"];
+                }
+                if (parameters.AllKeys.Contains("authorizationCode"))
+                {
+                    configSection.AuthorizationCode = parameters["authorizationCode"];
+                }
+                 if (parameters.AllKeys.Contains("redirectURI"))
+                {
+                    configSection.RedirectURI = parameters["redirectURI"];
+                }
+            }
+            
+            if (!configSection.ApplicationType.Equals("server"))
```

Comment:
I find the way to check the application type using the negated form a bit not natural. I would use the positive check for ApplicationType to be "public" or "web" and set the exception message to be "AuthorizationCode and RedirectURI are required. For Public/Web Apps, AuthCode and Redirect URI must be provided in config file or passed when instantiating ETClient"